test/new-e2e/tests/agent-metric-logs/linux-log-agent/01_file_tailing_test.go

```diff
@@ -0,0 +1,174 @@
+		if !assert.Empty(c, logs, "Logs were found when none were expected.") {
+			cat, _ := s.Env().VM.ExecuteWithError("cat /var/log/hello-world.log && cat /var/log/hello-world-2.log")
+			t.Logf("Logs detected when none were expected: %v", cat)
+		}
+	}, 2*time.Minute, 10*time.Second)
+
+	// Part 2: Adjust permissions of new log file before log generation
+	output, err := s.Env().VM.ExecuteWithError("sudo chmod +r /var/log/hello-world.log && echo true")
+	assert.NoError(t, err, "Unable to adjust permissions for the log file '/var/log/hello-world.log'.")
+	if strings.TrimSpace(output) == "true" {
+		t.Logf("Permissions granted for new log file.")
+		// Generate log
+		generateLog(s, "hello-before-permission-world", 1)
+
+		// Part 3: Check intake for new logs
+		checkLogs(s, "hello", "hello-before-permission-world", true)
+	}
+
+	// Part 3: Adjust permissions of new log file after log generation
+
+	// Restore permissions to default and generate log
+	output, err = s.Env().VM.ExecuteWithError("sudo chmod 644 /var/log/hello-world.log && echo true")
+	assert.NoError(t, err, "Unable to adjust back to default permissions for the log file '/var/log/hello-world.log'.")
+	if strings.TrimSpace(output) == "true" {
+		t.Logf("Permissions reset to default.")
+		generateLog(s, "hello-after-permission-world", 1)
+	}
+
+	// Grant log file permission and check intake for new logs
+	output, err = s.Env().VM.ExecuteWithError("sudo chmod +r /var/log/hello-world.log && echo true")
+	assert.NoError(t, err, "Unable to adjust permissions for the log file '/var/log/hello-world.log'.")
+	if strings.TrimSpace(output) == "true" {
+		t.Logf("Permissions granted for log file.")
+		checkLogs(s, "hello", "hello-after-permission-world", true)
+	}
+}
+
+func (s *LinuxVMFakeintakeSuite) LogPermission() {
+	t := s.T()
+	logPath := "/var/log/hello-world.log"
+	checkLogFilePresence(s, logPath)
+
+	// Part 4: Allow on only write permission to the log file so the agent cannot tail it
+	output, err := s.Env().VM.ExecuteWithError("sudo chmod -r /var/log/hello-world.log && echo true")
+	assert.NoError(t, err, "Unable to adjust permissions for the log file '/var/log/hello-world.log'.")
+	if strings.TrimSpace(output) == "true" {
+		t.Logf("Read permissions revoked")
+		s.Env().VM.Execute("sudo service datadog-agent restart")
+
+		generateLog(s, "access-denied", 1)
+		// Check intake to see if new logs are not present
+		checkLogs(s, "hello", "access-denied", false)
+	}
+
+	// Part 5: Restore permissions
+	output, err = s.Env().VM.ExecuteWithError("sudo chmod +r /var/log/hello-world.log && echo true")
+	assert.NoError(t, err, "Unable to adjust permissions for the log file '/var/log/hello-world.log'.")
+
+	// Part 6: Generate new logs and check the intake for new logs
+	if strings.TrimSpace(output) == "true" {
+		t.Logf("Permissions restored.")
+		generateLog(s, "access-granted", 1)
+		// Check intake to see if new logs are present
+		checkLogs(s, "hello", "access-granted", true)
```

Comment:
There is some delay between the attempts made by the agent to open the file. Do we need to add a `sleep` here to account for that delay? Or are we relying on the retry of `checkLogs`?